crates/zizmor/src/main.rs

```diff
@@ -955,7 +955,8 @@ async fn main() -> ExitCode {
 
                         Some(report)
                     }
-                    CollectionError::Yamlpath(..) => {
+                    // These errors only happen if something is wrong with zizmor itself.
+                    CollectionError::Yamlpath(..) | CollectionError::Model(..) => {
```

Comment:
Noting: this turns the somewhat confusing render in #1395 into a nice tip render, but I don't have a great way to test it (since these kinds of errors are exceptional by nature/indicate bugs that I try to fix).